### PR TITLE
Add config docs

### PR DESCRIPTION
**Pull request by phackstock**
This PR adds documentation for the new nomenclature configuration feature.

### CODE REVIEW

File: docs/user_guide.rst [hunk 1]
```diff
@@ -48,3 +48,4 @@ the RegionProcessor and validated using DataStructureDefinition.
   user_guide/model-mapping
   user_guide/model-registration
   user_guide/local-usage
+  user_guide/config
```
Comment (danielhuppmann) [inline docs/user_guide.rst:51]: I suggest to have "local-usage" last in this list.

File: docs/user_guide/config.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+.. _config:
+
+.. currentmodule:: nomenclature
+
+Nomenclature configuration
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:5]:
```suggestion
General project configuration
```

File: docs/user_guide/config.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+The configuration file **must be** located on the same level as the *mappings* and
+*definitions* directories.
+
+The file **must be** in yaml format and named *nomenclature.yaml*.
+
+Importing from an external repository
+-------------------------------------
+
+In order to import form an external repository the configuration file must define the
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:25]:
```suggestion
In order to import from an external repository, the configuration file must define the
```

File: docs/user_guide/config.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+repository und the repositories key.
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:26]:
```suggestion
repository and the repositories key.
```

File: docs/user_guide/config.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+The repository needs a name, in the example below *common-definitions* and a *url*, for
+example:
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:29]:
```suggestion
The repository has a **name** (in the example below *common-definitions*) and a **url**:
```

File: docs/user_guide/config.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+  repositories:
+    common-definitions:
+      url: https://github.com/IAMconsortium/common-definitions.git/
+  definitions:
+    region:
+      repository: common-definitions
+    variable:
+      repository: common-definitions
+
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:56]: Please add the *mappings* to the example.

File: docs/user_guide/config.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+In order for the import to work the url must be given in the above format, i.e. with the
+leading *https://* and the trailing *.git/*.
+
+Information from an external repository can either be used for codelists or model
+mappings, or both.
+
+For use in, for example, the *region* and *variable* codelists, the following is
+required:
+
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:45]: This sentence can be removed without loss of information.
```suggestion
```

File: docs/user_guide/config.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+.. code:: yaml
+
+  repositories:
+    common-definitions:
+      url: https://github.com/IAMconsortium/common-definitions.git/
+
+In order for the import to work the url must be given in the above format, i.e. with the
+leading *https://* and the trailing *.git/*.
+
+Information from an external repository can either be used for codelists or model
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:40]:
```suggestion
Information from an external repository can either be used for codelists ("definitions") or model
```

File: docs/user_guide/config.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  definitions:
+    region:
+      repository: common-definitions
+    variable:
+      repository: common-definitions
+
+The value in *definitions.region.repository* needs to reference the repository in the
+*repositories* section.
+
+Adding a common list of countries to the region codelist
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:60]:
```suggestion
Adding countries to the region codelist
```

File: docs/user_guide/config.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+By setting *definitions.region.country* as *true* in the configuration file:
+
+.. code:: yaml
+
+  definitions:
+    region:
+      country: true
+
+the nomenclature package will add a common list of countries to a region codelist.
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:71]:
```suggestion
the nomenclature package will add all countries to the *region* codelist.
```

**Review by danielhuppmann — COMMENTED** [review]
A few suggestions inline, and please also add the config file to the directory structure, see https://github.com/IAMconsortium/nomenclature/blob/b4cbb53c9b2e4681c25726c25f71f5b4b8e4637e/docs/user_guide/directory-structure.rst?plain=1#L23

File: docs/user_guide/config.rst (hunk 6 above)
Comment (phackstock) [reply docs/user_guide/config.rst:54]: done

File: docs/user_guide.rst (hunk 1 above)
Comment (phackstock) [reply docs/user_guide.rst:50]: done

File: docs/user_guide/config.rst [hunk 11]
```diff
@@ -0,0 +1,74 @@
+.. _config:
+
+.. currentmodule:: nomenclature
+
+General project configuration
+==========================
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:6]:
```suggestion
=============================
```

File: docs/user_guide/config.rst [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+    repository: common-definitions
+
+The value in *definitions.region.repository* needs to reference the repository in the
+*repositories* section.
+
+For model mappings the process is analogous using *mappings.repository*.
+
+
+Adding countries to the region codelist
+--------------------------------------------------------
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:62]:
```suggestion
---------------------------------------
```

File: docs/user_guide/config.rst [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+.. currentmodule:: nomenclature
+
+General project configuration
+==========================
+
+The nomenclature package features a configuration file that is used to enable the
+following features:
+
+* Import codelists from public GitHub repositories
+* Add a list of common countries (details: :ref:`countries`) to the region codelist
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:12]:
```suggestion
* Add all countries to the region codelist (details: :ref:`countries`) 
```

File: docs/user_guide/config.rst [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+.. currentmodule:: nomenclature
+
+General project configuration
+==========================
+
+The nomenclature package features a configuration file that is used to enable the
+following features:
+
+* Import codelists from public GitHub repositories
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:11]:
```suggestion
* Import codelists and mappings from public GitHub repositories
```

File: docs/user_guide/config.rst [hunk 15]
```diff
@@ -0,0 +1,74 @@
+.. _config:
+
+.. currentmodule:: nomenclature
+
+General project configuration
+==========================
+
+The nomenclature package features a configuration file that is used to enable the
+following features:
```
Comment (danielhuppmann) [inline docs/user_guide/config.rst:9]:
```suggestion
The following features can be accessed via a general project configuration file:
```

**Review by danielhuppmann — APPROVED** [review]
Looks good, a few final formatting and shortening suggestions, then good to merge and do a release!

**Comment by phackstock** [comment]
Thanks for the corrections @danielhuppmann, I'll go ahead with the merge and the new release.

**Pull request merged**